src/azul/service/repository_controller.py

```diff
@@ -61,7 +64,11 @@ def download_file(self,
             file_version = file['version']
             drs_path = file['drs_path']
             if file_name is None:
-                file_name = file['name']
+                # To prevent file overwrites due to files with the same name
+                # we insert the crc32c value of the file into the filename.
```

Comment:
```suggestion
                # To prevent overwriting one file with another of the same name
                # but different content
                # we insert the crc32c value of the file into the filename. This case frequently occurs in analysis bundles.
```